pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -353,12 +361,22 @@ func (nm *NodeManager) DiscoverNode(nodeID string, searchBy cm.FindVM) error {
 				foundExternal = true
 				break
 			}
+
+			if !foundInternal && !foundExternal {
+				return fmt.Errorf("unable to find suitable IP address for node %s with IP family %s", nodeID, ipFamily)
+			}
+
+			if _, ok := os.LookupEnv("ENABLE_ALPHA_DUAL_STACK"); ok {
```

Comment:
I think we want the alpha gate check to happen when checking the ipFamily field in the config.